pyomo/contrib/mpc/examples/cstr/run_mhe.py

```diff
@@ -129,68 +115,20 @@ def run_cstr_mhe(
     # Construct least square objective to minimize measurement errors
     # and model disturbances
     #
-    # This flag toggles between two different objective formulations.
-    # I included it just to demonstrate that we can support both.
-    error_var_objective = True
-    if error_var_objective:
-        error_vars = [
-            pyo.Reference(esti_blo.measurement_error_variables[idx, :])
-            for idx in esti_blo.measurement_set
-        ]
-        # This cost function penalizes the square of the "error variables"
-        m_estimator.measurement_error_cost = get_cost_from_error_variables(
-            error_vars, m_estimator.sample_points
-        )
-    else:
-        from pyomo.common.collections import ComponentMap
-        measurement_map = ComponentMap(
-            (var, [
-                esti_blo.measurement_variables[i, t]
-                for t in m_estimator.sample_points
-            ])
-            for i, var in enumerate(measured_variables)
-        )
-        setpoint_data = mpc.TimeSeriesData(
-            measurement_map, m_estimator.sample_points
-        )
-        # This cost function penalizes the difference between measurable
-        # estimates and their corresponding measurements.
-        error_cost = get_tracking_cost_from_time_varying_setpoint(
-            measured_variables, m_estimator.sample_points, setpoint_data
-        )
-        m_estimator.measurement_error_cost = error_cost
-
-    #
-    # Construct disturbance cost expression
-    #
-    disturbance_vars = [
-        pyo.Reference(esti_blo.disturbance_variables[idx, :])
-        for idx in esti_blo.disturbance_set
-    ]
-
-    # We know what order we sent constraints to the disturbance constraint
-    # function, so we know which indices correspond to which equations
-    weights = {
-        esti_blo.disturbance_variables[0, :]: 10.0,
-        esti_blo.disturbance_variables[1, :]: 10.0,
-    }
-    m_estimator.model_disturbance_cost = get_cost_from_error_variables(
-        disturbance_vars, m_estimator.sample_points, weight_data=weights
-    )
-    ###
-
-    m_estimator.squred_error_disturbance_objective = pyo.Objective(
-        expr=(sum(m_estimator.measurement_error_cost.values()) +
-              sum(m_estimator.model_disturbance_cost.values())
-              )
+    cost = estimator_spt_interface.get_tracking_cost_from_target_trajectory(
+        m_estimator.measurements,
+        variables=measured_variables,
     )
-
-    #
-    # Initialize measurements to initial values of measured variables
-    #
-    for index, var in enumerate(measured_variables):
-        for spt in m_estimator.sample_points:
-            esti_blo.measurement_variables[index, spt].set_value(var[spt].value)
```

Comment:
This initialization now happens when the measurement parameters are constructed.